### PR TITLE
Benji/ios 302 reactions

### PR DESCRIPTION
**Pull request by Benjamindodgson**

### CODE REVIEW

File: Shared/UI/Views/MessageContentView/MessageContentView+Menu.swift [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+//
+
+import Foundation
+
+#if IOS
+import StreamChat
+
+extension MessageContentView: UIContextMenuInteractionDelegate {
+
+    func contextMenuInteraction(_ interaction: UIContextMenuInteraction,
```
Comment (Benjamindodgson) [inline Shared/UI/Views/MessageContentView/MessageContentView+Menu.swift:16]: Add the context menu directly to the view.  Lot less hassle.

File: iOS/Flows/Conversation/Cells/ConversationMessageCellDatasource.swift [hunk 2]
```diff
@@ -78,9 +81,9 @@ extension ConversationMessageCellDataSource {
             // The menu interaction should only be on the front most cell.
             cell.content.backgroundColorView.interactions.removeAll()
             if zIndex == 0 {
-                cell.setContextMenuInteraction(with: item.contextMenuDelegate)
+                cell.content.setContextMenu()
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Cells/ConversationMessageCellDatasource.swift:84]: we can still set it if we want.

File: iOS/Flows/Conversation/Cells/ReactionsView.swift [hunk 3]
```diff
@@ -13,26 +13,55 @@ import StreamChat
...
     override func layoutSubviews() {
         super.layoutSubviews()
 
         self.imageView.expandToSuperviewSize()
+        self.label.setSize(withWidth: self.width)
+        self.label.centerOnXAndY()
+    }
+
+    #if IOS
+    func configure(with reactions: Set<ChatMessageReaction>) {
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Cells/ReactionsView.swift:44]: display a reaction

File: iOS/Flows/Conversation/Conversation List/ConversationListViewController+Extensions.swift [hunk 4]
```diff
@@ -23,20 +23,8 @@ extension ConversationListViewController {
             self.onSelectedMessage?(item.channelID, item.messageID)
         }
 
-        self.dataSource.handleDeletedConversation = { (conversation) in
-            Task {
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Conversation List/ConversationListViewController+Extensions.swift:27]: not used.

File: iOS/Flows/Conversation/Threads/ThreadViewController.swift [hunk 5]
```diff
@@ -267,6 +277,10 @@ extension ThreadViewController {
             self.conversationController?.sendKeystrokeEvent(completion: nil)
         }.store(in: &self.cancellables)
 
+        self.messageController.messageChangePublisher.mainSink { [unowned self] changes in
+            // TODO
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Threads/ThreadViewController.swift:281]: Not sure how to listen for reactions being added. Future PR

File: iOS/_Stream Temp/MessageController+Extensions.swift [hunk 6]
```diff
@@ -54,6 +54,21 @@ extension MessageController {
...
+                             extraData: extraData) { error in
+                if let e = error {
+                    continuation.resume(throwing: e)
+                } else {
+                    continuation.resume(returning: ())
+                }
+            }
+        })
+    }
+
```
Comment (MartinJibber) [inline iOS/_Stream Temp/MessageController+Extensions.swift:71]: Thanks for adding this

**Review by MartinJibber — APPROVED** [review]

**Pull request merged**